sparse/client.go

```diff
@@ -146,23 +149,28 @@ func (client *syncClient) sendHTTPRequest(method string, action string, interval
 	req.Header.Add("Accept", "application/json")
 
 	q := req.URL.Query()
-	q.Add("begin", strconv.FormatInt(interval.Begin, 10))
-	q.Add("end", strconv.FormatInt(interval.End, 10))
+	for k, v := range queries {
+		q.Add(k, v)
+	}
 	req.URL.RawQuery = q.Encode()
 
 	log.Debugf("method: %s, url with query string: %s, data len: %d", method, req.URL.String(), len(data))
 
 	return httpClient.Do(req)
 }
 
-func (client *syncClient) openServer() error {
+func (client *syncClient) openServer(directIO bool) error {
 	var err error
 	var resp *http.Response
 
 	timeStart := time.Now()
 	timeStop := timeStart.Add(time.Duration(client.timeout) * time.Second)
+	queries := make(map[string]string)
+	queries["begin"] = strconv.FormatInt(0, 10)
+	queries["end"] = strconv.FormatInt(client.fileSize, 10)
```

Comment:
Don't need to add these 2 entries. Since the server won't invoke `getQueryInterval` after receiving `open` calls.

Same for `func (client *syncClient) closeServer() {`

Reply:
updated

Reply:
OH, sorry. This suggestion should be applied to `closeServer` only. And you already found that.